### PR TITLE
Unsigned Thea Extrinsic

### LINKED ISSUE #668: make thea Extrinsics Unsigned & Feeless
The following Thea extrinsics are called by the Thea relayer:

- `approve_deposit`
- `set_thea_key_complete`
- `thea_queued_queued_public_key`
- `thea_key_rotation_complete`

All four extrinsics should be feeless, which means that even if an active relayer has no balance, the transaction should still be executed.

Which of the following is true?

 - [x] All four extrinsics should be unsigned.
 - [x] All four extrinsics should be feeless.
 - [x] The validate_unsigned macro should be used to validate the payload before adding it to the tx pool.

### PR DESCRIPTION
**Pull request by nakul1010**
Closes #668.

### CODE REVIEW

**Comment by nakul1010** [comment]
@felixfaisal the [index here](https://github.com/Polkadex-Substrate/thea/blob/66e1c6ed99e49d54e57eee4cfd3245e1299dd8cc/relayer/src/main.rs#L48) is taken from CLI and passed to Controller and writer, but the scripts for starting relayer 2 [here](RUST_LOG=info cargo +nightly run -- -s "pudding shadow witness enrich noodle peasant fan deputy member monitor curious expire" -x 19946 -p 1450 -b /ip4/127.0.0.1/tcp/26000/p2p/"$ID" 2>&1 | tee relayer_2.log) and starting relayer 3 [here](https://github.com/Polkadex-Substrate/thea/blob/66e1c6ed99e49d54e57eee4cfd3245e1299dd8cc/relayer/start_relayer_3.sh#L7) doesn't contain the any CLI flag so all of the 3 relayers will have index zero right or is the index getting updated somewhere in the code.

File: pallets/thea/src/lib.rs [hunk 1]
```diff
@@ -389,17 +536,23 @@ pub mod pallet {
...
 			origin: OriginFor<T>,
 			bit_map: u128,
 			bls_signature: [u8; 96],
 			token_type: TokenType,
 			payload: Vec<u8>,
-		) -> DispatchResult {
-			let _relayer = ensure_signed(origin)?;
+			_relayer_index: u8,
+			_network: Network,
+			_signature: T::Signature,
```
Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs]: Why do we need these args if we are not using them?

File: pallets/thea/src/lib.rs [hunk 2]
```diff
@@ -530,6 +683,8 @@ pub mod pallet {
 		/// * `tx_hash`: Transaction hash of key update on foreign chain
 		/// * `bit_map`: Bitmap of Thea relayers
 		/// * `bls_signature`: BLS signature of relayers
+		/// * `_relayer_index`: The index of relayer in active set.
```
Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs]: Arg without underscore

File: pallets/thea/src/lib.rs [hunk 3]
```diff
@@ -538,8 +693,11 @@ pub mod pallet {
 			tx_hash: sp_core::H256,
 			bit_map: u128,
 			bls_signature: [u8; 96],
-		) -> DispatchResult {
-			let relayer = ensure_signed(origin)?;
+			relayer_index: u8,
+			_signature: T::Signature,
```
Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs:700]: Why do we need this arg if we are not using it?

File: pallets/thea/src/lib.rs [hunk 4]
```diff
@@ -589,6 +747,8 @@ pub mod pallet {
 		/// * `public_key`: New Public Key for thea (Raw Uncompressed)
 		/// * `bit_map`: Bitmap of Thea relayers
 		/// * `bls_signature`: BLS signature of relayers
+		/// * `_relayer_index`: The index of relayer in active set.
```
Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs]: Remove underscore at the beginning

File: pallets/thea/src/lib.rs [hunk 5]
```diff
@@ -597,8 +757,11 @@ pub mod pallet {
 			public_key: [u8; 64],
 			bit_map: u128,
 			bls_signature: [u8; 96],
-		) -> DispatchResult {
-			let relayer = ensure_signed(origin)?;
+			relayer_index: u8,
+			_signature: T::Signature,
```
Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs:765]: Why do we need this args if we are not using it?

File: pallets/thea/src/lib.rs [hunk 6]
```diff
@@ -647,8 +812,10 @@ pub mod pallet {
 			public_key: [u8; 64],
 			bit_map: u128,
 			bls_signature: [u8; 96],
-		) -> DispatchResult {
-			let _relayer = ensure_signed(origin)?;
+			_relayer_index: u8,
+			_signature: T::Signature,
```
Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs:822]: Why do we need these args if we are not using them?

File: pallets/thea/src/lib.rs [hunk 7]
```diff
@@ -954,6 +1121,33 @@ pub mod pallet {
 
 	// Helper Functions for Thea Pallet
 	impl<T: Config> Pallet<T> {
+		/// Retrieve the AccountId of the relayer account at the specified index for the specified
+		/// network id
+		///
+		/// Arguments:
```
Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs]: Please use common `# Parameters` here for the consistency purpose.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/thea/src/lib.rs (hunk 1 above)
Comment (nakul1010) [reply pallets/thea/src/lib.rs]: Yes, we are using the argument to validate the tx, check [here](https://github.com/Polkadex-Substrate/Polkadex/pull/669/files#diff-db9aab8a8240e8c00955d7f7debad560f2c408d932d7ab7bbbb94575855c5dadR471)

File: pallets/thea/src/lib.rs (hunk 3 above)
Comment (nakul1010) [reply pallets/thea/src/lib.rs:700]: Yes, we are using the argument to validate the tx

File: pallets/thea/src/lib.rs (hunk 5 above)
Comment (nakul1010) [reply pallets/thea/src/lib.rs:765]: Yes, we are using the argument to validate the tx

File: pallets/thea/src/lib.rs (hunk 6 above)
Comment (nakul1010) [reply pallets/thea/src/lib.rs:822]: Yes, we are using the argument to validate the tx

File: pallets/thea/src/lib.rs (hunk 1 above)
Comment (serhii-temchenko) [reply pallets/thea/src/lib.rs]: Could you please add comment in dockblock of each method used those fields to not remove them at some point and understand the purpose of these fields?

File: pallets/thea/src/lib.rs (hunk 3 above)
Comment (serhii-temchenko) [reply pallets/thea/src/lib.rs:700]: Same as [here](https://github.com/Polkadex-Substrate/Polkadex/pull/669#discussion_r1143054287)

File: pallets/thea/src/lib.rs (hunk 5 above)
Comment (serhii-temchenko) [reply pallets/thea/src/lib.rs:765]: Same as [here](https://github.com/Polkadex-Substrate/Polkadex/pull/669#discussion_r1143054287)

File: pallets/thea/src/lib.rs (hunk 6 above)
Comment (serhii-temchenko) [reply pallets/thea/src/lib.rs:822]: Same as [here](https://github.com/Polkadex-Substrate/Polkadex/pull/669#discussion_r1143054287)

**Review by serhii-temchenko — APPROVED** [review]

**Comment by nakul1010** [comment]
End to End Test failing

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**